### PR TITLE
feat: Added Virtual Network Gateways and Connections

### PR DESCRIPTION
**Pull request by mlozoya2**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

This PR adds `Virtual Network Gateways` and `Virtual Network Gateway Connections` as related tables to `Virtual Networks`.

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [x] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [x] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

**Review by yevgenypats — APPROVED** [review]
Thank you! LGTM.

Will also give @roneli to review.

**Comment by erezrokah** [comment]
Thanks for the PR @mlozoya2. Tests are failing since we verify the resources gets extracted. For that to work we need to add those to our Terraform deployments in https://github.com/cloudquery/cq-provider-azure/tree/a9b524e98440b094bba2ec009a78b857a0f90b24/terraform.

I'll take a look at adding those

File: resources/services/network/virtual_networks.go [hunk 1]
```diff
@@ -435,6 +435,450 @@ func NetworkVirtualNetworks() *schema.Table {
...
+						Description: "VPN authentication types for the virtual network gateway.",
+						Type:        schema.TypeStringArray,
+						Resolver:    schema.PathResolver("VirtualNetworkGatewayPropertiesFormat.VpnClientConfiguration.VpnAuthenticationTypes"),
+					},
+					{
+						Name:        "vpn_client_configuration_vpn_client_ipsec_policies",
+						Description: "VpnClientIpsecPolicies for virtual network gateway P2S client.",
+						Type:        schema.TypeJSON,
+						Resolver:    schema.PathResolver("VirtualNetworkGatewayPropertiesFormat.VpnClientConfiguration.VpnClientIpsecPolicies"),
+					},
```
Comment (roneli) [inline resources/services/network/virtual_networks.go:566]: nit: would maybe remove the redundant vpn_client in some of the columns

**Review by roneli — APPROVED** [review]
LGTM, had one nit

**Review by roneli — APPROVED** [review]

**Comment by erezrokah** [comment]
Thanks @mlozoya2 this is released in [v0.11.7](https://github.com/cloudquery/cq-provider-azure/releases/tag/v0.11.7) 🚀

**Pull request merged**